lms/services/jstor.py

```diff
@@ -62,28 +115,46 @@ def _get_public_url(self, url):
         :raises ExternalRequestError: If we get bad data back from the service
         """
 
-        doi = url.replace("jstor://", "")
-        if "/" not in doi:
-            doi = f"{self.DEFAULT_DOI_PREFIX}/{doi}"
+        article_id = url.replace("jstor://", "")
+        doi = self._doi_from_article_id(article_id)
+        s3_url = self._api_request(format_uri("/pdf-url/{}", doi)).text
 
+        if not s3_url.startswith("https://"):
+            raise ExternalRequestError(
+                f"Expected to get an S3 URL but got: '{s3_url}' instead"
+            )
+
+        return s3_url
+
+    def _api_request(self, endpoint: str) -> requests.Response:
+        """
+        Make an authenticated request to the JSTOR API.
+
+        See the JSTOR API's `/docs` endpoint for details.
+        """
         token = JWTService.encode_with_secret(
             {"site_code": self._site_code},
             secret=self._secret,
             lifetime=timedelta(hours=1),
         )
+        url = self._api_url + endpoint
+        return self._http.get(url=url, headers={"Authorization": f"Bearer {token}"})
 
-        s3_url = self._http.request(
-            method="GET",
-            url=f"{self._api_url}/pdf-url/{doi}",
-            headers={"Authorization": f"Bearer {token}"},
-        ).text
+    def _doi_from_article_id(self, article_id):
+        if "/" not in article_id:
+            return f"{self.DEFAULT_DOI_PREFIX}/{article_id}"
+        return article_id
 
-        if not s3_url.startswith("https://"):
-            raise ExternalRequestError(
-                f"Expected to get an S3 URL but got: '{s3_url}' instead"
-            )
 
-        return s3_url
+def format_uri(template: str, *params: str):
```

Comment:
TIL about the typing of args

https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values